src/ctap2/mod.rs

```diff
@@ -531,19 +532,35 @@ pub fn sign<Dev: FidoDevice>(
         }
     }
 
+    let mut allow_list = args.allow_list;
+    let mut rp_id = args.relying_party_id;
+    let client_data_hash = ClientDataHash(args.client_data_hash);
+    if let Some(alt_rp_id) = args.alternate_rp_id {
+        if !allow_list.is_empty() {
+            // Try to silently discover U2F credentials that require the FIDO App ID extension. If
+            // any are found, we should use the alternate RP ID instead of the provided RP ID.
+            let silent_creds = silently_discover_credentials(
```

Comment:
Why do we do this here again? Didn't we already do this in `StateMachine::sign()`?
Couldn't we reuse the `silent_creds`-list we acquired in `StateMachine`, and simply overwrite the `args.allow_list` and `args.relying_party_id` there, before handing it in?

Reply:
StateMachine is only used by USB HID devices, and I intend to write virtual device tests for this.